mtags/src/main/scala-2/scala/meta/internal/pc/AutoImports.scala

```diff
@@ -60,11 +60,40 @@ trait AutoImports { this: MetalsGlobal =>
             )
           }
 
-        def forScript =
-          for {
-            obj <- lastVisitedParentTrees.collectFirst { case mod: ModuleDef =>
-              mod
+        def firstBodyLineOffset(implPos: Position): Int = {
+          @tailrec
+          def loop(index: Int): Int = {
+            if (index >= text.length()) index
+            else if (text.charAt(index) == '\n' && index + 1 < text.length())
+              index + 1
+            else loop(index + 1)
+          }
+          val firstBodyLine = loop(implPos.start)
+          // skip comments/ using directives
+          val additionalOffest =
+            text.drop(firstBodyLine).offsetAfterUsingDirectives
+
+          firstBodyLine + additionalOffest
+        }
+
+        def forScript = {
+          println(text)
+          println("~~~")
+          val moduleDef =
+            if (text.startsWith("package ammonite")) {
+              // ammonite wraps sc code into `object test { ... }`
+              lastVisitedParentTrees.collect {
+                case mod: ModuleDef if mod.name.decoded == "test" => mod
```

Comment:
This is not true, Ammonite wraps `object foo{` if the script name is `foo.sc` (backtick if it is required)

https://github.com/com-lihaoyi/Ammonite/blob/bec02ae120eca3002b19405ddaea71b1fbbcaed9/amm/compiler/src/main/scala/ammonite/compiler/DefaultCodeWrapper.scala#L24
(Ammonite script runner looks like provides the file name for it).

Reply:
ouch. thx for finding this!